src/components/views/rooms/UserIdentityWarning.tsx

```diff
@@ -0,0 +1,321 @@
+    // discard the value that we fetched.  We always use the value from the
+    // update and consider it as the most up-to-date version.  If the fetched
+    // value is more up-to-date, then we should be getting a new update soon
+    // with the newer value, so it will fix itself in the end.
+    const gotVerificationStatusUpdateRef = useRef<Map<string, boolean>>(new Map());
+
+    // Select a new user to display a warning for.  This is called after the
+    // current prompted user no longer needs their identity approved.
+    const selectCurrentPrompt = useCallback((): RoomMember | undefined => {
+        const membersNeedingApproval = membersNeedingApprovalRef.current;
+        if (membersNeedingApproval.size === 0) {
+            return undefined;
+        }
+
+        // We pick the user with the smallest user ID.
+        const keys = Array.from(membersNeedingApproval.keys()).sort((a, b) => a.localeCompare(b));
+        const selection = membersNeedingApproval.get(keys[0]!);
+        return selection;
+    }, []);
+
+    // Add a user to the membersNeedingApproval map, and update the current
+    // prompt if necessary. The user will only be added if they are actually a
+    // member of the room. If they are not a member, this function will do
+    // nothing.
+    const addMemberNeedingApproval = useCallback(
+        (userId: string, member?: RoomMember): void => {
+            if (userId === cli.getUserId()) {
+                // We always skip our own user, because we can't pin our own identity.
+                return;
+            }
+            member = member ?? room.getMember(userId) ?? undefined;
+            if (member) {
+                membersNeedingApprovalRef.current.set(userId, member);
+                // We only select the prompt if we are done initialising,
+                // because we will select the prompt after we're done
+                // initialising, and we want to start by displaying a warning
+                // for the user with the smallest ID.
+                if (initialisedRef.current === InitialisationStatus.Completed) {
+                    setCurrentPrompt((currentPrompt) => {
+                        // If we aren't currently displaying a warning, we pick
+                        // a new user to show a warning for.  If we are already
+                        // displaying a warning, don't change the display.
+                        //
+                        // We have to do this in a callback to
+                        // `setCurrentPrompt` because this function could have
+                        // been called after an `await`, and the `currentPrompt`
+                        // that this function would have may be outdated.
+                        if (!currentPrompt) {
+                            return selectCurrentPrompt();
+                        } else {
+                            return currentPrompt;
+                        }
+                    });
```

Comment:
suggest hoisting the comment out of the callback.

```suggestion
                    // If we aren't currently displaying a warning, we pick
                    // a new user to show a warning for.  If we are already
                    // displaying a warning, don't change the display.
                    //
                    // We have to do this in a callback to `setCurrentPrompt`,
                    // because this function could have been called after an 
                    // `await`, and the `currentPrompt` that this function 
                    // would have may be outdated.
                    setCurrentPrompt((currentPrompt) => currentPrompt ?? selectCurrentPrompt());
```